configure.ac

```diff
@@ -66,14 +66,21 @@ AC_ARG_ENABLE([urpc],
     [enable_urpc=$enableval],
     [enable_urpc=no]
 )
-AC_ARG_ENABLE([perfflow],
-    [AS_HELP_STRING([--enable-perfflow],
+AM_CONDITIONAL([URPC], [test "x$enable_urpc" = "xyes"])
+AC_ARG_WITH([perfflow],
+    [AS_HELP_STRING([--with-perfflow],
                     [enable performance measurement with PerfFlow Aspect])],
-    [enable_perfflow=$enableval],
-    [enable_perfflow=no]
+    [with_perfflow=$withval],
+    [with_perfflow=no]
+)
+AM_CONDITIONAL([PERFFLOW], [test "x$with_perfflow" = "xyes"])
+AC_ARG_WITH([ucx],
+    [AS_HELP_STRING([--with-ucx],
+                    [build the UCX-based data transport layer])],
+    [with_ucx=$withval],
+    [with_ucx=no]
 )
-# TODO Add support for libb64 back once base64 encoding/decoding is fully complete
-# AC_ARG_VAR([LIBB64_DIR], [root directory for libb64])
+AM_CONDITIONAL([UCX], [test "x$with_ucx" = "xyes"])
```

Comment:
What is the purpose of having both with_ucx and enable ucx both having yes/no values. 

Generally --with-ucx should take the path right? and --enable-ucx would be just enabling it.

Reply:
I'm not sure what you mean. There is no `--enable-ucx` in `configure.ac`

Reply:
I know generally u wanna say u have an option called --enable-ucx 
and optionally u can set path to ucx with an option called --with-ucx

Reply:
I see. That makes more sense. I missed that characteristic of `with-package`. I'll switch to `AC_ARG_ENABLE` for these. I can add `AC_ARG_WITH` before taking the PR out of draft.